lib/monitoring/metrics/authenticator_gauge.rb

```diff
@@ -0,0 +1,61 @@
+module Monitoring
+  module Metrics
+    class AuthenticatorGauge
+      attr_reader :registry, :pubsub, :metric_name, :docstring, :labels, :sub_event_name
+
+      def setup(registry, pubsub)
+        @registry = registry
+        @pubsub = pubsub
+        @metric_name = :conjur_server_authenticator
+        @docstring = 'Number of authenticators enabled'
+        @labels = [:type, :status]
+        @sub_event_name = 'conjur.policy_loaded'
+        
+        # Create/register the metric
+        Metrics.create_metric(self, :gauge)
+
+        # Run update to set the initial counts on startup
+        update
+      end
+
+      def update(*payload)
+        metric = @registry.get(@metric_name)
+        update_enabled_authenticators(metric)
+        update_installed_authenticators(metric)
+        update_configured_authenticators(metric)
+      end
+
+      def update_enabled_authenticators(metric)
+        enabled_authenticators = Authentication::InstalledAuthenticators.enabled_authenticators
```

Comment:
I'd recommend using dependency injection to make `Authentication::InstalledAuthenticators` available inside the class.  This will greatly simplify unit testing.

Something like:

```ruby 
module Monitoring
  module Metrics
    class AuthenticatorGauge < Monitoring::Metrics::BaseMetric      
      def initialize(installed_authenticators: Authentication::InstalledAuthenticators)
        @metric_name = :conjur_server_authenticator
        @docstring = 'Number of authenticators enabled'
        @labels = [:type, :status]
        @sub_event_name = 'conjur.policy_loaded'

        @installed_authenticators = installed_authenticators
      end

      ...

    end
  end
end
```